src/lib/Microsoft.Health.Extensions.Fhir.R4/BundleExtensions.cs

```diff
@@ -41,21 +41,28 @@ public static IEnumerable<TResource> ReadFromBundle<TResource>(this Bundle bundl
             }
         }
 
-        public static TResource ReadOneFromBundle<TResource>(this Bundle bundle, bool throwOnMultipleFound = true)
+        public static async Task<TResource> ReadOneFromBundleWithContinuationAsync<TResource>(this Bundle bundle, IFhirClient fhirClient, bool throwOnMultipleFound = true)
             where TResource : Resource, new()
         {
-            var bundleCount = bundle?.Entry?.Count ?? 0;
-            if (bundleCount == 0)
+            if (bundle == null)
+            {
+                return null;
+            }
+
+            var resources = await bundle?.ReadFromBundleWithContinuationAsync<TResource>(fhirClient, 2);
+
+            var resourceCount = resources.Count();

```

Comment:
Minor, but best to use .ToArray to avoid multiple enumerations over the same collection (happens with Count & FirstOrDefault).  Impact is minimal since ReadFromBundleWithContinuation is using a materialized collection behind the scenes (list in this case) but if that ever changes we should re-evaluate.  

We also may want to look at using async enumerables going forward as well https://docs.microsoft.com/en-us/archive/msdn-magazine/2019/november/csharp-iterating-with-async-enumerables-in-csharp-8